docker-app/qfieldcloud/core/views/users_views.py

```diff
@@ -146,3 +172,185 @@ class ListUserOrganizationsView(generics.ListAPIView):
+
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+class TeamDetailView(APIView):
+
+    def get(self, request, organization_name, team_name):
+        """
+        GET /organizations/<str:organization_name>/team/<str:team_name>/
+        Retrieve the details of a specific team within the organization.
+        """
+        organization = get_object_or_404(Organization, username=organization_name)
+        team_name = f'@{organization_name}/{team_name}'
+        team = get_object_or_404(Team, team_organization=organization, username=team_name)
+
+        serializer = TeamAccessSerializer(team, data={'request': request})
+        
+        serializer.is_valid(raise_exception=True)
+        
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+    def put(self, request, organization_name, team_name):
+        """
+        PUT /organizations/<str:organization_name>/team/<str:team_name>/
+        Update a team's name within the organization.
+        """
+        organization = get_object_or_404(Organization, username=organization_name)
+        team_name = f'@{organization_name}/{team_name}'
+        team = get_object_or_404(Team, team_organization=organization, username=team_name)
+
+        new_team_name = request.data.get('username')
+        
+        if not new_team_name:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        if Team.objects.filter(team_organization=organization, username=new_team_name).exists():
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        new_team_name = f'@{organization_name}/{new_team_name}'
+        team.username = new_team_name
+        
+        team.save()
+
+        serializer = TeamSerializer(team)
+        
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def delete(self, request, organization_name, team_name):
+        """
+        DELETE /organizations/<str:organization_name>/team/<str:team_name>/
+        Delete a team within the organization.
+        """
+        organization = get_object_or_404(Organization, username=organization_name)
+        team_name = f'@{organization_name}/{team_name}'
+        team = get_object_or_404(Team, team_organization=organization, username=team_name)
+
+        team.delete()
+        
+        serializer = TeamDeleteSerializer(data={'message': "Team deleted successfully"})
+        serializer.is_valid(raise_exception=True)
+        
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class TeamListCreateView(APIView):
+
+    def get(self, request, organization_name):
+        """
+        Handle GET request: Retrieve all teams under a specific organization --> organizations/<str:organization_name>/team/
+        """
+        organization = get_object_or_404(Organization, username=organization_name)
+
+        teams = querysets_utils.get_organization_teams(organization)
+
+        serializer = TeamListSerializer(teams, many=True)
```

Comment:
```suggestion
        organization = get_object_or_404(Organization, username=organization_name)
        teams = querysets_utils.get_organization_teams(organization)
        serializer = TeamListSerializer(teams, many=True)
```

It's all variable assignment, so no need fow the newlines. Anyways, this most probably goes away.

Reply:
Done